### ISSUE TITLE
Docs about gnome-keyring assume Arch/Pacman, should have a Debian/Ubuntu reference also

### ISSUE THREAD

**Issue by aitor-gomila** [microsoft/vscode#176320]

**Comment by TylerLeonhardt** [comment]
Actually... do you use the WSL extension? https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl

With that extension, the secrets are actually stored on the _client_ side... so the Windows side. This is much cleaner as keyrings (like `gnome-keyring`) usually expect a GUI... and with WSL it's a little wonky.

**Comment by aitor-gomila** [comment]
> Actually... do you use the WSL extension? https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl
> 
> With that extension, the secrets are actually stored on the _client_ side... so the Windows side. This is much cleaner as keyrings (like `gnome-keyring`) usually expect a GUI... and with WSL it's a little wonky.

I use it now as a workaround, but I prefer to have all my tooling in the WSL side, if I ever need to move to an actual machine.

**Comment by TylerLeonhardt** [comment]
You didn't have a GUI with a window manager then, yeah? You were relying on Windows to run the Linux VS Code GUI?

**Comment by aitor-gomila** [comment]
I was using WSLg

**Comment by TylerLeonhardt** [comment]
Yeah so for WSLg (isn't that provided by Windows?), there's not a real window manager AFAIK, like no gnome or whatever. Keyrings are provided by the window manager (gnome-keyring, kwallet, etc) so in order to get this working we need to convince the container (WSL) that it has everything it needs for one of these keyrings to work.

I'm sure this is possible, maybe with dbus, but it involves some trial and error.

**Comment by aitor-gomila** [comment]
It's a wayland backend called weston and a few other things, no GNOME or KDE though, though it shouldn't matter.
